addons/mail/static/src/components/message/message.xml

```diff
@@ -180,6 +180,9 @@
                             </ul>
                         </t>
                     </div>
+                    <t t-if="threadViewer and message.originThread and message.originThread !== threadViewer.thread and message.subject">
+                        <p class="o_Message_subject">Subject: <t t-esc="message.subject"/></p>
+                    </t>
```

Comment:
Ideally subject should always be displayed when relevant...

...But the main 2 reasons for not displaying this piece of info are 1) irrelevant subject value (= name of mailing channel or chatter's record) and 2) ugly UI (insertion after body in italics apparently the "best" solution so far...).

With this change, subject is always displayed on inbox, history. Messages with subject that are viewer in context of a non-origin thread will also always display the subject, which means:

1. a channel that is follower of (one or many) mailing channel(s) will always display the subject
2. a channel that is follower of (one or many) chatter's record(s) will always display the subject

The 2nd aspect is the most problematic, as it adds too much noise.
Considering all issues, I would be as conservative as possible and just display subjects in inbox and history. Ignore (mailing) channels for the moment, because this introduces issues that are more annoying than just not having them.

Reply:
So basically a t-if with "thread.id = mail.inbox or thread.id = mail.history" ?

Reply:
yeah, kinda